packages/matter-node.js/src/matter/MatterController.ts

```diff
@@ -36,32 +38,48 @@ const ADMIN_VENDOR_ID = new VendorId(752);
 const logger = Logger.get("MatterController");
 
 export class MatterController {
-    public static async create(scanner: Scanner, netInterfaceIpv4: NetInterface, netInterfaceIpv6: NetInterface) {
-        const certificateManager = new RootCertificateManager();
+    public static async create(scanner: Scanner, netInterfaceIpv4: NetInterface, netInterfaceIpv6: NetInterface, persistenceManager: PersistenceManager) {
+        const certificateManager = new RootCertificateManager(persistenceManager);
+
         const ipkValue = Crypto.getRandomData(16);
         const fabricBuilder = new FabricBuilder(FABRIC_INDEX)
             .setRootCert(certificateManager.getRootCert())
             .setRootNodeId(CONTROLLER_NODE_ID)
             .setIdentityProtectionKey(ipkValue)
             .setRootVendorId(ADMIN_VENDOR_ID);
         fabricBuilder.setOperationalCert(certificateManager.generateNoc(fabricBuilder.getPublicKey(), FABRIC_ID, CONTROLLER_NODE_ID));
-        const fabric = await fabricBuilder.build();
-        return new MatterController(scanner, netInterfaceIpv4, netInterfaceIpv6, certificateManager, fabric);
+
+        // Check if we have a fabric stored in the persistence, if yes initialize this one, else build a new one
+        const controllerPersistance = persistenceManager.createPersistence("MatterController");
+        if (controllerPersistance.has("fabric")) {
+            const storedFabric = Fabric.fromJson(controllerPersistance.get<FabricJsonObject>("fabric"));
+            return new MatterController(scanner, netInterfaceIpv4, netInterfaceIpv6, certificateManager, storedFabric, persistenceManager);
+        } else {
+            return new MatterController(scanner, netInterfaceIpv4, netInterfaceIpv6, certificateManager, await fabricBuilder.build(), persistenceManager);
+        }
     }
 
-    private readonly sessionManager = new SessionManager(this);
+    private readonly sessionManager;
     private readonly channelManager = new ChannelManager();
-    private readonly exchangeManager = new ExchangeManager<MatterController>(this.sessionManager, this.channelManager);
+    private readonly exchangeManager;
     private readonly paseClient = new PaseClient();
     private readonly caseClient = new CaseClient();
+    private readonly controllerPersistence: Persistence;
 
     constructor(
         private readonly scanner: Scanner,
         private readonly netInterfaceIpv4: NetInterface,
         private readonly netInterfaceIpv6: NetInterface,
         private readonly certificateManager: RootCertificateManager,
         private readonly fabric: Fabric,
+        private readonly persistenceManager: PersistenceManager
     ) {
+        this.controllerPersistence = this.persistenceManager.createPersistence("MatterController");
+
+        this.sessionManager = new SessionManager(this, this.persistenceManager);
```

Comment:
Couldn't manager use `this` to ask for storageManager?

Reply:
Yes, if we define the persistenceManager as public ...  Do we want that? ( I know JS do not care that much, but OOP wise it i smeeh in my eyes to have it public just for that, or?)

Reply:
There could be a getter for it.

Reply:
Ok, I will adjust and add a public getter

Reply:
The sessionmanager is using "this" as a generic Context and is used by Device and Controller, but in fact Sessionmanager do not know what it really is typing wise. I would like to leave it that way for now ...